openpype/hosts/max/plugins/publish/validate_frame_range.py

```diff
@@ -0,0 +1,60 @@
+import pyblish.api
+
+from pymxs import runtime as rt
+from openpype.pipeline import (
+    OptionalPyblishPluginMixin
+)
+from openpype.pipeline.publish import (
+    RepairAction,
+    ValidateContentsOrder,
+    PublishValidationError
+)
+
+
+class ValidateFrameRange(pyblish.api.InstancePlugin,
+                         OptionalPyblishPluginMixin):
+    """Validates the frame ranges.
+
+    This is an optional validator checking if the frame range on instance
+    matches the frame range specified for the asset.
+
+    It also validates render frame ranges of render layers.
+
+    Repair action will change everything to match the asset frame range.
+
+    This can be turned off by the artist to allow custom ranges.
+    """
+
+    label = "Validate Frame Range"
+    order = ValidateContentsOrder
+    families = ["maxrender"]
+    hosts = ["max"]
+    optional = False
+    actions = [RepairAction]
+
+    def process(self, instance):
+        if not self.is_active(instance.data):
+            self.log.info("Skipping validation...")
+            return
+        context = instance.context
+
+        frame_start = int(context.data.get("frameStart"))
+        frame_end = int(context.data.get("frameEnd"))
+
+        inst_frame_start = int(instance.data.get("frameStart"))
+        inst_frame_end = int(instance.data.get("frameEnd"))
+
+        if frame_start != inst_frame_start:
+            raise PublishValidationError(
+                "startFrame on instance does not match"
+                " with startFrame from the context data")
+
+        if frame_end != inst_frame_end:
+            raise PublishValidationError(
+                "endFrame on instance does not match"
+                " with endFrame from the context data")
```

Comment:
This should actually report the values too so the user knows what to do.
Additionally I feel it's weird that it only reports the start frame differences if both start frame and end frame differ. I'd expect to get a report about both start and end frame if they are wrong, especially because repair would also be changing both.

Reply:
There have been changes. I'd say however my comment here is still unresolved currently. In essence:

- Show both start / end frame mismatches instead of only reporting start frame mismatch if both differed.
- I'd say that it should include the current value and the expected value in the message too.